src/core/StelTexture.cpp

```diff
@@ -204,13 +207,14 @@ void StelTexture::waitForLoaded()
 		loader->waitForFinished();
 }
 
-template <typename T, typename Param, typename Arg>
-void StelTexture::startAsyncLoader(T (*functionPointer)(Param), const Arg &arg)
-{
-	Q_ASSERT(loader==Q_NULLPTR);
-	//own thread pool only supported with Qt 5.4+
-	loader = new QFuture<GLData>(QtConcurrent::run(textureMgr->loaderThreadPool, functionPointer, arg));
-}
+// GZ Sorry I just cannot get the right template formulation... We need it only 2x, so made explicit.
```

Comment:
Why not simply remove this? This template doesn't seem useful anyway, and even if it were, it's still there in the revision history.

Reply:
If anyone of you knows how to fix it, we could fix it. Else sure, if you agree, remove before merge. Maybe it's trivial, just I don't see it.

Reply:
Well, the fix seems to be simple: just add one more parameter to the function (with template type), or better, make the template variadic to take any number of parameters to pass. But IMO this template function is just an unnecessary complication, so I'd simply remove it.

Reply:
I added the Arg2 argument to the startAsyncLoader template function, but it did not help. Or I forgot something. But where???

@alex-w do you agree that I can just remove the template?

Reply:
You forgot to modify the `functionPointer` signature. It should take two parameters:
```cpp
template <typename T, typename Param1, typename Param2, typename Arg1, typename Arg2>
void StelTexture::startAsyncLoader(T (*functionPointer)(Param1,Param2), const Arg1 &arg, const Arg2& arg2)
```
But I'd use a more general approach, if we do leave this function here:
```cpp
template <typename T, typename...Params, typename...Args>
void StelTexture::startAsyncLoader(T (*functionPointer)(Params...), Args&&...args)
{
    Q_ASSERT(loader==Q_NULLPTR);
    //own thread pool only supported with Qt 5.4+
    loader = new QFuture<GLData>(QtConcurrent::run(textureMgr->loaderThreadPool, functionPointer,
                                                   std::forward<Args>(args)...));
}
```

Reply:
d'oh yes I see it now... 
And esp. thank you for the full solution.